### PR TITLE
feat: Create win 2025 images for CNI and CNS using windows servercore insider tags.

### PR DESCRIPTION
**Pull request by sivakami-projects**
**Reason for Change**:

Create window 2025 images for CNI and CNS using the insider tags provided by windows, until ltsc2025 tags are officially release by windows servercore.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: dropgz/build/windows.Dockerfile [hunk 1]
```diff
@@ -35,5 +35,6 @@ COPY --from=compressor /dropgz .
 RUN GOOS=windows CGO_ENABLED=0 go build -a -o bin/dropgz.exe -trimpath -ldflags "-X github.com/Azure/azure-container-networking/dropgz/internal/buildinfo.Version="$VERSION"" -gcflags="-dwarflocationlists=true" main.go
 
 FROM mcr.microsoft.com/windows/nanoserver:${OS_VERSION} as windows
+
```
Comment (jpayne3506) [inline dropgz/build/windows.Dockerfile:38]: nit: extra space

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 .DEFAULT_GOAL := help
-
+# Test
```
Comment (jpayne3506) [inline Makefile:2]: Plans to remove this?

File: Makefile [hunk 3]
```diff
@@ -22,8 +22,7 @@ GOOSES   ?= "linux windows" # To override at the cli do: GOOSES="\"darwin bsd\""
 GOARCHES ?= "amd64 arm64" # To override at the cli do: GOARCHES="\"ppc64 mips\""
 ltsc2019  = "10.0.17763.4010"
 ltsc2022  = "10.0.20348.643"
-ltsc2022-KB5035857   = "10.0.26100.1"
-win2025 = "10.0.26100.1"
+win2025  = "10.0.26244.5000"
```
Comment (jpayne3506) [inline Makefile:25]: Lets go ahead and change this to `ltsc2025` and make the appropriate changes in the pipeline and docker files.

File: cni/Dockerfile [hunk 4]
```diff
@@ -43,6 +46,12 @@ FROM scratch as linux
...
 ENTRYPOINT [ "/dropgz" ]
 
-FROM mcr.microsoft.com/windows/nanoserver:${OS_VERSION} as windows
+FROM mcr.microsoft.com/windows/nanoserver:ltsc2019 as ltsc2019
+
+FROM mcr.microsoft.com/windows/nanoserver:ltsc2022 as ltsc2022
+
+FROM mcr.microsoft.com/windows/nanoserver/insider:win2025 as win2025
+
+FROM ${OS_VERSION} as windows
```
Comment (jpayne3506) [inline cni/Dockerfile:55]: Lets pin these to a sha.

**Review by jpayne3506 — CHANGES_REQUESTED** [review]
Will discuss stateless changes offline.

File: Makefile (hunk 3 above)
Comment (sivakami-projects) [reply Makefile:25]: version used for win2025 is not officially GA'ed yet. Should we "ltsc" prefix when it is not GA'ed?

Comment (paulyufan2) [reply Makefile:24]: Hi John, I think we can use win2025 or similar name until win2025 is officially released.

Comment (rbtr) [reply Makefile:24]: if you call it `ltsc2025` now, you only need to update the value of this variable in this once place when it is released for real. 
if you call it `win2025` now, when it is released you will need to update value AND the variable name everywhere.
the intent of this work is to support ltsc2025, right? call it that.

Comment (rbtr) [reply Makefile:24]: and fix the alignment

File: cni/Dockerfile [hunk 5]
```diff
@@ -2,16 +2,19 @@ ARG ARCH
...
 ARG OS
 ARG VERSION
+ARG STATELESS
 WORKDIR /azure-container-networking
 COPY . .
 RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/network/plugin/main.go
 RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet-telemetry -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/telemetry/service/telemetrymain.go
 RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet-ipam -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/ipam/plugin/main.go
 RUN GOOS=$OS CGO_ENABLED=0 go build -a -o /go/bin/azure-vnet-stateless -trimpath -ldflags "-X main.version="$VERSION"" -gcflags="-dwarflocationlists=true" cni/network/stateless/main.go
+RUN if [ "$STATELESS" = "true" ]; then mv /go/bin/azure-vnet-stateless /go/bin/azure-vnet; fi
```
Comment (rbtr) [inline cni/Dockerfile:17]: don't do this

File: Makefile [hunk 6]
```diff
@@ -1,5 +1,4 @@
 .DEFAULT_GOAL := help
-
```
Comment (rbtr) [inline Makefile:2]: why?

File: cni/Dockerfile [hunk 7]
```diff
@@ -43,6 +46,12 @@ FROM scratch as linux
 COPY --from=dropgz /go/bin/dropgz dropgz
 ENTRYPOINT [ "/dropgz" ]
 
-FROM mcr.microsoft.com/windows/nanoserver:${OS_VERSION} as windows
+FROM mcr.microsoft.com/windows/nanoserver@sha256:736f8845da1c7bbc9510b419abddb870230485e74644a96e0001a21e1ca4c172 as ltsc2019
+
+FROM mcr.microsoft.com/windows/nanoserver@sha256:cc9f6a1334190c1f18a905f56d285132e562a9a1ce18ff9476cfaf95194aa0b7 as ltsc2022
+
+FROM mcr.microsoft.com/windows/nanoserver/insider@sha256:67e0ab7f3a79cd73be4a18bae24659c03b294aed0dbeaa624feb3810931f0bd2 as win2025
```
Comment (jpayne3506) [inline cni/Dockerfile:53]: Add a comment above each intermediate layer.

**Review by jpayne3506 — APPROVED** [review]

**Review by paulyufan2 — APPROVED** [review]

**Review by rbtr — APPROVED** [review]

**Pull request merged**